src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -86,7 +86,7 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
         return builder.AddParameter(
                 new ParameterResource(
                     name,
-                    parameterDefault => parameterDefault != null ? parameterDefault.GetDefaultValue() : valueGetter(),
+                    parameterDefault => parameterDefault?.GetDefaultValue() ?? valueGetter(),
```

Comment:
This is a behavior change. Previously if `parameterDefault.GetDefaultValue()` returned `null`, `null` would be returned and valueGetter wouldn't have been invoked.

Reply:
Ah true, will revert.

Reply:
That said, it's marked as non-nullable, so not sure the behavior is well-defined if `GetDefaultValue()` returns null.

Reply:
Yeah, relooking at the code, I don't think this would be a problem in reality.

Reply:
Still, won't make the change here as it's unrelated to this fix.